src/abstractsynchronizer.js

```diff
@@ -247,13 +246,17 @@ olcs.AbstractSynchronizer.prototype.listenForGroupChanges_ = function(group) {
       ];
       listenKeyArray.push.apply(listenKeyArray, contentKeys);
     }
-  }, this);
+  }).bind(this);
 
   listenAddRemove();
 
   listenKeyArray.push(group.on('change:layers', function(e) {
     contentKeys.forEach(function(el) {
-      goog.array.remove(listenKeyArray, el);
+      var i = listenKeyArray.indexOf(el);
+      goog.asserts.assert(i >= 0);
+      if (i >= 0) {
```

Comment:
The reason is that `splice()` accepts negative argument, which is definitely not what we want.
Having both the assertion and the `if` allows to always have a correct behaviour and both catch the issue at debug time.